client-src/elements/chromedash-feature-table.ts

```diff
@@ -1,52 +1,49 @@
 import {LitElement, css, html, nothing} from 'lit';
-import {showToastMessage, clamp} from './utils.js';
+import {customElement, property, state} from 'lit/decorators.js';
+import {SHARED_STYLES} from '../css/shared-css.js';
+import {Feature} from '../js-src/cs-client.js';
 import './chromedash-feature-filter';
 import './chromedash-feature-row';
-import {SHARED_STYLES} from '../css/shared-css.js';
+import {clamp, showToastMessage} from './utils.js';
 
+@customElement('chromedash-feature-table')
 class ChromedashFeatureTable extends LitElement {
-  static get properties() {
-    return {
-      query: {type: String},
-      showEnterprise: {type: Boolean},
-      sortSpec: {type: String},
-      showQuery: {type: Boolean},
-      features: {type: Array},
-      totalCount: {type: Number},
-      loading: {type: Boolean},
-      reloading: {type: Boolean},
-      start: {type: Number},
-      num: {type: Number},
-      alwaysOfferPagination: {type: Boolean},
-      columns: {type: String},
-      signedIn: {type: Boolean},
-      canEdit: {type: Boolean},
-      starredFeatures: {type: Object},
-      noResultsMessage: {type: String},
-      gates: {type: Object},
-      selectedGateId: {type: Number},
-    };
-  }
-
-  constructor() {
-    super();
-    this.query = '';
-    this.showEnterprise = false;
-    this.sortSpec = '';
-    this.showQuery = false;
-    this.loading = true;
-    this.reloading = false;
-    this.starredFeatures = new Set();
-    this.features = [];
-    this.totalCount = 0;
-    this.start = 0;
-    this.num = 100;
-    this.alwaysOfferPagination = false;
-    this.noResultsMessage = 'No results';
-    this.canEdit = false;
-    this.gates = {};
-    this.selectedGateId = 0;
-  }
+  @state()
+  query = '';
+  @state()
+  loading = true;
+  @state()
+  reloading = false;
+  @state()
+  features!: Feature[];
+  @state()
+  totalCount = 0;
+  @state()
+  start = 0;
+  @property({type: Boolean})
+  showEnterprise = false;
+  @property({type: String})
+  sortSpec = '';
+  @property({type: Boolean})
+  showQuery = false;
+  @property({type: Object})
+  starredFeatures = new Set<number>();
+  @property({type: Number})
+  num = 100;
+  @property({type: Boolean})
+  alwaysOfferPagination = false;
+  @property({type: String})
+  noResultsMessage = 'No results';
+  @property({type: Boolean})
+  canEdit = false;
+  @property({type: Object})
+  gates: any; // TODO(markxiong0122): Add Gate type when PR#4060 is merged.
+  @property({type: Number})
+  selectedGateId = 0;
+  @property({type: String})
+  columns;
+  @property({type: Boolean})
+  signedIn;
```

Comment:
Ideally, these would include `: boolean` or `: string` so that uses get type-checked. (`columns` might even be `"normal"|"approvals"` rather than `string`, but I haven't checked how it's used.)